test/python/circuit/test_circuit_load_from_qpy.py

```diff
@@ -0,0 +1,258 @@
+        qpy_file = io.BytesIO()
+        dump(qpy_file, qc)
+        qpy_file.seek(0)
+        new_circ = load(qpy_file)[0]
+        self.assertEqual(qc, new_circ)
+
+    def test_custom_gate(self):
+        """Test that custom  gate is correctly serialized"""
+        custom_gate = Gate('black_box', 1, [])
+        custom_definition = QuantumCircuit(1)
+        custom_definition.h(0)
+        custom_definition.rz(1.5, 0)
+        custom_definition.sdg(0)
+        custom_gate.definition = custom_definition
+
+        qc = QuantumCircuit(1)
+        qc.append(custom_gate, [0])
+        qpy_file = io.BytesIO()
+        dump(qpy_file, qc)
+        qpy_file.seek(0)
+        new_circ = load(qpy_file)[0]
+        self.assertEqual(qc, new_circ)
+        self.assertEqual(qc.decompose(), new_circ.decompose())
+
+    def test_custom_instruction(self):
+        """Test that custom instruction is correctly serialized"""
+        custom_gate = Instruction('black_box', 1, 0, [])
+        custom_definition = QuantumCircuit(1)
+        custom_definition.h(0)
+        custom_definition.rz(1.5, 0)
+        custom_definition.sdg(0)
+        custom_gate.definition = custom_definition
+        qc = QuantumCircuit(1)
+        qc.append(custom_gate, [0])
+        qpy_file = io.BytesIO()
+        dump(qpy_file, qc)
+        qpy_file.seek(0)
+        new_circ = load(qpy_file)[0]
+        self.assertEqual(qc, new_circ)
+        self.assertEqual(qc.decompose(), new_circ.decompose())
+
+    def test_parameter(self):
+        """Test that a circuit with a parameter is correctly serialized."""
+        theta = Parameter('theta')
+        qc = QuantumCircuit(5, 1)
+        qc.h(0)
+        for i in range(4):
+            qc.cx(i, i + 1)
+
+        qc.barrier()
+        qc.rz(theta, range(5))
+        qc.barrier()
+        for i in reversed(range(4)):
+            qc.cx(i, i + 1)
+        qc.h(0)
+        qc.measure(0, 0)
+
+        qpy_file = io.BytesIO()
+        dump(qpy_file, qc)
+        qpy_file.seek(0)
+        new_circ = load(qpy_file)[0]
+        self.assertEqual(qc, new_circ)
+        self.assertEqual(qc.bind_parameters({theta: 3.14}),
+                         new_circ.bind_parameters({theta: 3.14}))
+
+    def test_bound_parameter(self):
+        """Test a circuit with a bound parameter is correctly serialized."""
+        theta = Parameter('theta')
+        qc = QuantumCircuit(5, 1)
+        qc.h(0)
+        for i in range(4):
+            qc.cx(i, i + 1)
+
+        qc.barrier()
+        qc.rz(theta, range(5))
+        qc.barrier()
+        for i in reversed(range(4)):
+            qc.cx(i, i + 1)
+        qc.h(0)
+        qc.measure(0, 0)
+        qc.assign_parameters({theta: 3.14})
+
+        qpy_file = io.BytesIO()
+        dump(qpy_file, qc)
+        qpy_file.seek(0)
+        new_circ = load(qpy_file)[0]
+        self.assertEqual(qc, new_circ)
+
+    def test_parameter_expression(self):
+        """Test a circuit with parameter expression raises."""
```

Comment:
This docstring should be updated since parameter expressions are supported now.

Reply:
Good catch, fixed in: https://github.com/Qiskit/qiskit-terra/pull/5578/commits/b09cec8081b3c66e75a1f6dd23964a72a314dcac